src/Symfony/Component/Form/Extension/Core/Type/FileType.php

```diff
@@ -46,19 +46,16 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     }
                 }
 
-                // submitted data for an input file (not required) without choosing any file
-                if (array(null) === $data || array() === $data) {
+                // Since the array is never considered empty in the view data format
+                // on submission, we need to evaluate the configured empty data here
+                if (array() === $data) {
                     $emptyData = $form->getConfig()->getEmptyData();
-
-                    $data = is_callable($emptyData) ? call_user_func($emptyData, $form, $data) : $emptyData;
+                    $data = $emptyData instanceof \Closure ? $emptyData($form, $data) : $emptyData;
                 }
 
                 $event->setData($data);
             } elseif (!$requestHandler->isFileUpload($event->getData())) {
-                $emptyData = $form->getConfig()->getEmptyData();
-
-                $data = is_callable($emptyData) ? call_user_func($emptyData, $form, $data) : $emptyData;
-                $event->setData($data);
+                $event->setData(null);
```

Comment:
And that this is a bc break introduced in #24993 with does not count as the one which adds a method to the `RequestHanlderInterface` introduced in that same PR, considering the [intended behavior](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Form/Form.php#L615).